src/decoder/flac.rs

```diff
@@ -1,62 +1,302 @@
+//! - **Channels**: Supports mono, stereo, and multi-channel audio (up to 8 channels)
+//! - **Seeking**: Full forward and backward seeking with sample-accurate positioning
+//! - **Duration**: Accurate total duration calculation from stream metadata
+//! - **Performance**: Optimized block-based decoding with reusable buffers
+//!
+//! # Advantages
+//!
+//! - **Perfect quality**: Lossless compression preserves original audio fidelity
+//! - **Efficient compression**: Typically 30-50% size reduction vs. uncompressed
+//! - **Fast decoding**: Optimized algorithms with minimal computational overhead
+//! - **Precise seeking**: Sample-accurate positioning without approximation
+//! - **Rich metadata**: Comprehensive format specification with extensive metadata support
+//!
+//! # Limitations
+//!
+//! - Seeking requires `is_seekable` setting for backward seeks (forward-only otherwise)
+//! - No support for embedded cue sheets or complex metadata (focus on audio data)
+//! - Larger files than lossy formats (trade-off for perfect quality)
+//!
+//! # Configuration
+//!
+//! The decoder can be configured through `DecoderBuilder`:
+//! - `with_seekable(true)` - Enable backward seeking (recommended for FLAC)
+//! - Other settings are informational and don't affect FLAC decoding performance
+//!
+//! # Performance Notes
+//!
+//! - Block-based decoding minimizes memory allocations during playback
+//! - Seeking operations use efficient linear scanning or stream reset
+//! - Buffer reuse optimizes performance for continuous playback
+//! - Metadata parsing optimized for audio-focused applications
+//! - Memory usage scales with maximum block size, not file size
+//!
+//! # Example
+//!
+//! ```ignore
+//! use std::fs::File;
+//! use rodio::Decoder;
+//!
+//! let file = File::open("audio.flac").unwrap();
+//! let decoder = Decoder::builder()
+//!     .with_data(file)
+//!     .with_seekable(true)
+//!     .build()
+//!     .unwrap();
+//!
+//! // FLAC supports seeking and bit depth detection
+//! println!("Bit depth: {:?}", decoder.bits_per_sample());
+//! println!("Duration: {:?}", decoder.total_duration());
+//! println!("Sample rate: {}", decoder.sample_rate().get());
+//! ```
 
-use crate::source::SeekError;
-use crate::Source;
+use std::{
+    io::{Read, Seek},
+    mem,
+    sync::Arc,
+    time::Duration,
+};
 
-use crate::common::{ChannelCount, Sample, SampleRate};
-
-use claxon::FlacReader;
+use claxon::{FlacReader, FlacReaderOptions};
 use dasp_sample::Sample as _;
 use dasp_sample::I24;
 
-/// Decoder for the FLAC format.
+use super::{utils, Settings};
+use crate::{
+    common::{ChannelCount, Sample, SampleRate},
+    source::SeekError,
+    Source,
+};
+
+/// Reader options for `claxon` FLAC decoder.
+///
+/// Configured to skip metadata parsing and vorbis comments for faster initialization.
+/// This improves decoder creation performance by only parsing essential stream information
+/// needed for audio playback.
+///
+/// # Fields
```

Comment:
why not document the fields? themselves?

Reply:
Rustdoc doesn't support documenting fields in a constant.